linkerd/app/inbound/src/direct.rs

```diff
@@ -238,13 +242,22 @@ impl TryFrom<(tls::ConditionalServerTls, listen::Addrs)> for ClientInfo {
     }
 }
 
+impl ClientInfo {
+    fn header_negotiated(&self) -> bool {
```

Comment:
nit/TIOLI: i might prefer 
```suggestion
    fn is_header_negotiated(&self) -> bool {
```